main.go

```diff
@@ -178,6 +187,8 @@ func initFlags(fs *pflag.FlagSet) {
 		"",
 		"Set custom service endpoint in semi-colon separated format: ${ServiceRegion1}:${ServiceID1}=${URL1},${ServiceID2}=${URL2};${ServiceRegion2}:${ServiceID1}=${URL1}",
 	)
+
+	flags.AddDiagnosticsOptions(fs, &diagnosticsOptions)
```

Comment:
> this shouldn't be removed like this, lets deprecate it and remove in the next release with an appropriete TODO

@mkumatag, this has been removed above because it is being [handled here ](https://github.com/kubernetes-sigs/cluster-api/blob/main/util/flags/diagnostics.go#L42-L55)